### PR TITLE
Publish device Registered / Unregistered events to the EventBus

### LINKED ISSUE #392: Publish device Registered / Unregistered events to the EventBus
### What is the purpose of the feature?
Our application using plgd is event-driven, subscribed directly to NATS events. We're able to find all resource-related events except device `REGISTERED` and `UNREGISTERED` events as they are defined [here](https://github.com/plgd-dev/cloud/blob/v2/grpc-gateway/pb/events.proto#L18).

### What specific challenge does this feature solve?
Building a fully event-driven application on top of plgd. Currently, as 2 events are missing it's not possible.

### What alternatives did I consider?
We could pull information from the authorization server, but such a design doesn't meet our requirements.

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: authorization/pb/events.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package ocf.cloud.auth.pb;
+
+option go_package = "github.com/plgd-dev/cloud/authorization/pb;pb";
+
+// devices was registered to owner.
+message RegisteredDevices {
```
Comment (ondrejtomcik) [inline authorization/pb/events.proto]: The event described what has happened.
`DevicesRegistered`and `DevicesUnregistered`.

File: authorization/pb/events.proto [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    string owner = 1; // determined by clients.storage.ownerClaim.
+    repeated string device_ids = 2; // list of device ids added to owner.
+    int64 timestamp = 3; // unix timestamp in nanoseconds of creation event.
```
Comment (ondrejtomcik) [inline authorization/pb/events.proto:16]: I miss our AuditContext, which is helpful if somebody is building an audit log and you want to track who shared the device with this new owner.

File: authorization/pb/events.proto [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// devices was deregistered from owner.
+message UnregisteredDevices {
+    string owner = 1; // determined by clients.storage.ownerClaim
+    repeated string device_ids = 2; // list of device ids deleted from owner.
+    int64 timestamp = 3; // unix timestamp in nanoseconds of creation event.
+}
+
+// nats: owners.{owner}.>
+message Event {
```
Comment (ondrejtomcik) [inline authorization/pb/events.proto:29]: For resource-related events, we don't have such an envelope. Developers working with NATS won't like two different approaches. I would organize event types per subjects, as we do with RA.

Comment (jkralik) [reply authorization/pb/events.proto:29]: No. As a developer sometimes I want to be subscribed at  `owners.{owner}.>` to get all events. If I subscribe lower then I need to get the subject that provided the event. And I don't want to push the dependence to nats to the protocol.

Comment (ondrejtomcik) [reply authorization/pb/events.proto:29]: It's okay to organize it per as you defined -> `owners.{owner}.>`. My point is that `DevicesRegistered` shall be pushed to `owners.{owner}.deviceregistered` and `DevicesUnregistered` to `owners.{owner}.deviceunregistered`. This follows existing concepts you already introduced in RA.

**Review by Danielius1922 — APPROVED** [review]

**Review by Danielius1922 — CHANGES_REQUESTED** [review]
Static analysis failed

**Review by ondrejtomcik — APPROVED** [review]

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**